File: Data Structures & Algorithms/longest-consecutive-sequence/submission-22.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        if len(nums) == 0:
            return 0

        seen = set()

        for num in nums:
            seen.add(num)
        
        longestSequence = 1

        for num in nums:
            currentSequence = 1
            prev = num - 1
            while prev in seen:
                currentSequence += 1
                prev -= 1
            longestSequence = max(longestSequence, currentSequence)
        
        return longestSequence

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-22.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max = 0;

        int i = 0;
        while (i < prices.size()) {
            int j = i+1;
            while (j < prices.size() and prices[j] > prices[i]) {
                max = std::max(max, prices[j]-prices[i]);
                j++;
            }
            i = j;
        }
        return max;
    }
};
